src/gradualizer_lib.erl

```diff
@@ -83,58 +84,86 @@ reverse_graph(G) ->
     from_edges(maps:keys(G), [ {J, I} || {I, Js} <- maps:to_list(G), J <- Js ]).
 
 
+get_user_type_definition(Types, Anno, Name, Args) ->
+    %% Let's check if the type is a known remote type.
+    case typelib:get_module_from_annotation(Anno) of
+        {ok, Module} ->
+            case gradualizer_db:get_opaque_type(Module, Name, Args) of
+                {ok, Ty} ->
+                    {ok, Ty};
+                not_found ->
+                    not_found
+            end;
+        none ->
+            %% Let's check if the type is defined in the context of this module.
+            case maps:get({Name, length(Args)}, Types, not_found) of
+                {_Params, Ty} ->
+                    {ok, Ty};
+                not_found ->
+                    not_found
+            end
+    end.
+
+
 % Given a type, pick a value of that type.
 % Used in exhaustiveness checking to show an example value
 % which is not covered by the cases.
-pick_value(List) when is_list(List) ->
-    [pick_value(Ty) || Ty <- List ];
-pick_value(?type(integer)) ->
+pick_value(Types, List) when is_list(List) ->
```

Comment:
I'd prefer to reverse the order of parameters, i.e. put Types param after the Type we're picking a value from. Also add some short comment about what the params are (or a spec) please.